### ISSUE TITLE
New maintainer(s) needed

### ISSUE THREAD

**Issue by mikey179** [bovigo/vfsStream#175]
As some might have noticed this project needs some updates to ensure it will work seemlessly with future versions of Composer and probably PHP as well. Also, there are several issues which are now open for quite some time without progess from my side. Unfortunately, I‘m lacking the time to take care of this and if I‘m honest the motivation isn‘t there either. Some have provided pull requests to tackle issues, but I‘m not comfortable simply accepting them without thinking about them properly. 

Therefore I think it‘s time to hand this project over to new maintainers. If you are interested in becoming a maintainer please comment below. To ensure the project continues properly I‘d like if suggested maintainers see support by other users of the project, e.g. by +1 their comment, so this project doesn‘t run in a situation like certain Node modules. Also, I‘d be willing to support new maintainers in case of questions regarding the project for a couple of months to make sure they get a good understanding of the project.

If you have any other suggestions regarding the topic please feel free to comment as well.

**Comment by jaapio** [comment]
Hi,
I am one of the core contributors of phpdocumentor. We are using your awesome work in our tests quite often. I never looked at the code of this project. But would be happy to help reviewing the pr's and answering issues.
Time is limited so I cannot take over the full project :pensive:
Let me know if I can help.

**Comment by bizurkur** [comment]
I'm interested and am looking for projects to keep me busy. Currently I just work on a mini framework ([Bitty](https://github.com/bittyphp)) that I don't think anyone but me uses.

I think vfsStream is awesome and there's definitely some tests I couldn't have done as easily without it, but I've never actually looked at the src until today.

I'm heavily focused on (obsessed with?) standards and testing and if I took over my first goals would be:
- Upgrade PHPUnit
- Add PHP 7.3 to the travis build
- Add an official coding standard (I personally favor the strictness of the Slevomat standard)
- Add mutation testing to help test any possibly untested things
- Add PHPStan to help catch any potential bugs
- Attempt to test the untested `trigger_warning` code and others to try for 100% coverage
- Start focusing on 2.0. This would include the obvious basics of:
  - Fix the composer package name so it doesn't break in composer 2
  - Move it to an organization repo (I believe one of the issues you had mentioned wanting to remove the "mikey179" part)
  - Start working on issues/open PRs.

I would probably have some questions in the beginning, but not too many.

**Comment by rokclimb15** [comment]
I'm also willing to help maintain this project. Please let me know how I can help. I would like to fix the composer 2.0 package name issue first and foremost.

**Comment by allejo** [comment]
I'll put my name forward to help out. Frankly, I've never looked at the source for this project but have used it often enough to know it's awesome and something I'd definitely be interested in learning how it works and would be happy to help maintain.

I like to stay active in open source projects to keep me from getting bored. I used [scrivo/highlight.php](https://github.com/scrivo/highlight.php) in one of my projects and since it was dead, I ended up maintaining it.

**Comment by mikey179** [comment]
Thanks everyone for offering your help and time to keep this project going. I‘m currently on vacation, but will be back next week and plan to spend some time on this. In the meantime, I got a question for you: as we have more than one offer I was thinking about if we could have more than one maintainer in the future so that there’s always one who could respond to issues etc. Would you be willing to be part of such a team? I guess this would involve setting up some guidelines how the team should work together and a COC. (Disclaimer: haven’t done this before. If you have experience with those topics and might be able to help with that but don’t want to necessarily work on the project itself I‘d like to hear from you.)

**Comment by bizurkur** [comment]
More maintainers / team = 👍 

I've never personally maintained a widely used library, so if someone else has more experience with that I'm all for them being the "lead" if one is needed. Otherwise I'd vote to implement https://github.com/Roave/BackwardCompatibilityCheck as part of the CI to prevent BC breaks.

**Comment by allejo** [comment]
I'm all for having multiple maintainers or a team, however you'd like to call it. Having a good number of eyes will help with splitting up tasks and reviewing code 👍

**Comment by rokclimb15** [comment]
@mikey179 I'm +1 for a team as well. I'm very partial to the [Ubuntu CoC](https://launchpad.net/codeofconduct/1.0.1) if you're looking for inspiration.

**Comment by mikey179** [comment]
I just suggested a CoC in #176. Please head over to the PR and let me know what you think.

As for the next steps: I think about moving this repository into the [bovigo](https://github.com/bovigo) org which I created some years ago. I‘d also move [bovigo/assert](https://github.com/mikey179/bovigo-assert) and [bovigo/callmap](https://github.com/mikey179/bovigo-callmap) into this org as vfsStream‘s test depend on those. With this org we could harness all the features Github provides.

**Comment by jaapio** [comment]
I like the idea of moving the repo to a github organisation. That will allow the new maintainers to invite people and remove the dependency of a single developer :+1: (in theory)

The CoC is very common, I don't have any issues or comments with it. 

Think we should not wait to long, and just get things going. If @mikey179 invites the people he would like to join the team in the new org. And moves the repo, so people who have energy to invest in the project can get started. We have some momentum now :-D

**Comment by mikey179** [comment]
Ok, I merged the CoC PR, moved all repositories including [vfs-stream-examples](https://github.com/bovigo/vfs-stream-examples) to the [bovigo](https://github.com/bovigo) organisation. Additionally, I branched of v1.x to create the just published version 1.6.6 with just the package name fix. Fixes for the current version should be done in this branch, while work on the next major version should be done in master (well, actually and prefered with pull requests I suppose).

In the next few minutes I'll invite everyone to the bovigo org who offered their support in this issue. Feel free to join or to ignore if you changed your mind. :)

In the next few days I'll try to sort through the current issues and update the roadmap for a possible 2.0 release. I think it's about time to finally drop support for PHP 5.x (at least in a new release, it's already dropped on master), and update to the latest and greatest versions of the dev dependencies for PHP ^7.2.
